src/Orchard.Web/Modules/Orchard.Widgets/Drivers/WidgetPartDriver.cs

```diff
@@ -35,26 +36,27 @@ public class WidgetPartDriver : ContentPartDriver<WidgetPart> {
                              () => shapeHelper.EditorTemplate(TemplateName: "Parts.Widgets.WidgetPart", Model: widgetPart, Prefix: Prefix))
             };
 
-            if (widgetPart.Id > 0)
-                results.Add(ContentShape("Widget_DeleteButton",
-                    deleteButton => deleteButton));
+            if (widgetPart.Id > 0 && widgetPart.TypeDefinition.Settings.GetModel<ContentTypeSettings>().Draftable) {
```

Comment:
Being able to unpublish does/should not depend on Draftable.
```suggestion
            if (widgetPart.Id > 0) {
```

Reply:
This is not addressed.